### PR TITLE
Feature/freespace service

### PR DESCRIPTION
**Pull request by marrts**
- Updated some meshes that were very large to reduce the lag it caused in RVIZ

- Added various meshes for environment and robot

- Updated launch files for simulation environment for both CRS workspace and mock SwRI workspace

- Created free-space server that either returns and can execute a free-space motion on the robot given a target Cartesian position through a custom service call

### CODE REVIEW

File: crs_motion_planning/src/freespace_planning_server.cpp [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+{
+public:
+  PlanningServer()
+    : Node("planning_server_node")
+  {
+    plan_service_ = this->create_service<crs_msgs::srv::CallFreespaceMotion>("test_plan", std::bind(&PlanningServer::planService, this, std::placeholders::_1, std::placeholders::_2));
+    traj_publisher_ = this->create_publisher<trajectory_msgs::msg::JointTrajectory>("set_trajectory_test",10);
+    joint_state_listener_ = this->create_subscription<sensor_msgs::msg::JointState>("/joint_states", 2, std::bind(&PlanningServer::jointCallback, this, std::placeholders::_1));
+
+    joint_name_list_ = {"shoulder_pan_joint", "shoulder_lift_joint", "elbow_joint", "wrist_1_joint", "wrist_2_joint", "wrist_3_joint"};
```
Comment (jrgnicho) [inline crs_motion_planning/src/freespace_planning_server.cpp]: It'd be nice to get this list from a parameter or the RobotModel class if there's one in ROS2

File: crs_motion_planning/src/freespace_planning_server.cpp [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+    std::string urdf_path, srdf_path;
+    urdf_path = ament_index_cpp::get_package_share_directory("crs_support") + "/urdf/crs.urdf";
```
Comment (jrgnicho) [inline crs_motion_planning/src/freespace_planning_server.cpp]: Same for these two variables, they should be loaded from parameters

File: crs_motion_planning/src/freespace_planning_server.cpp [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+private:
+  void jointCallback(const sensor_msgs::msg::JointState::SharedPtr joint_msg)
+  {
+    curr_joint_state_ = *joint_msg;
+  }
+  void planService(std::shared_ptr<crs_msgs::srv::CallFreespaceMotion::Request> request,
+                    std::shared_ptr<crs_msgs::srv::CallFreespaceMotion::Response> response)
+  {
+    std::string target_frame = request->target_link;
+    std::string base_link_frame = "world";
```
Comment (jrgnicho) [inline crs_motion_planning/src/freespace_planning_server.cpp]: The "world" string should be in a global const variable or loaded from a parameters

File: crs_motion_planning/src/freespace_planning_server.cpp [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    joint_init_waypoint->setIsCritical(false);
+
+    // Define goal waypoint
+    Eigen::Vector3d goal_pose(request->goal_pose.translation.x, request->goal_pose.translation.y, request->goal_pose.translation.z);
+    Eigen::Quaterniond goal_ori(request->goal_pose.rotation.w, request->goal_pose.rotation.x, request->goal_pose.rotation.y, request->goal_pose.rotation.z);
+    tesseract_motion_planners::CartesianWaypoint::Ptr goal_waypoint = std::make_shared<tesseract_motion_planners::CartesianWaypoint>(goal_pose, goal_ori);
+
+    goal_waypoint->setIsCritical(true);
+    Eigen::VectorXd coeffs(6);
+    coeffs << 10, 10, 10, 10, 10, 10;
```
Comment (jrgnicho) [inline crs_motion_planning/src/freespace_planning_server.cpp]: assign these coefficient values to a global const variable(s)

File: crs_motion_planning/src/freespace_planning_server.cpp [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+    // Add waypoints
+    trgt_wypts.push_back(joint_init_waypoint);
+    trgt_wypts.push_back(goal_waypoint);
+
+    // Write target waypoints to trajopt planning config
+    traj_pc->target_waypoints = trgt_wypts;
+
+    // Various freespace configuration settings
+    traj_pc->num_steps = 200;
```
Comment (jrgnicho) [inline crs_motion_planning/src/freespace_planning_server.cpp]: I think we may want to expose these values as configurable ros parameters in order to reconfigure without needing to recompile

File: crs_msgs/srv/CallFreespaceMotion.srv [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Target link
+string target_link
+# Target pose/ori
+geometry_msgs/Transform goal_pose
+# bool execute
```
Comment (jrgnicho) [inline crs_msgs/srv/CallFreespaceMotion.srv]: delete commented line

**Review by jrgnicho — CHANGES_REQUESTED** [review]

File: crs_motion_planning/src/freespace_planning_server.cpp (hunk 1 above)
Comment (marrts) [reply crs_motion_planning/src/freespace_planning_server.cpp]: I now do a lookup through tesseract

**Review by jrgnicho — APPROVED** [review]

**Comment by jrgnicho** [comment]
@marrts thanks for the contribution

**Pull request merged**